lib/rototiller/task/rototiller_task.rb

```diff
@@ -132,17 +133,19 @@ def set_verbose(verbosity=true)
         @verbose = verbosity
       end
 
-      def add_param(collection, param_class, param_array, args, &block)
+      def add_param(collection, param_class, param_array, args, opts={}, &block)
 
         if block_given?
 
           param_hash = pull_params_from_block(param_array, &block).to_h
+          param_hash[:set_env] = true if opts[:set_env]
           collection.push(param_class.new(param_hash))
         else
```

Comment:
seems like this method is now doing two very distinct things.  should be broken up